src/battlearena/SearchPlayers.js

```diff
@@ -0,0 +1,93 @@
+//@flow strict
+
+import * as React from 'react';
+
+type Props = {|
+	playerList: Array<string>,
+|};
+
+type State = {|
+	searchName: string,
+	searchList: Array<string>
+|};
+
+// Search Players Component. Takes an array of all players.
+//
+// Prop Names:
+// playerList (Array of all users)
+//
+// State Names:
+// searchName (the keyword the user is searching with)
+// searchList (list of all users that match searchName)
+// 
+// EXAMPLE PROP USAGE = <SearchPlayers playerList={ ["jeffery", "jeff", "john", "illeatbutt", "childeatbigtime", "childconsumer"] }/>
+//
+// Needs to be updated to have a challenge player button.
+class SearchPlayers extends React.Component<Props, State> {
+
+	constructor(props: Props) {
+		super(props);
+
+		this.state = {
+			searchName: "",
+			searchList: []
+		}
+	}
+
+	// When typing in the searchbar, update searchName.
+	handleChange = (e: string) => {
+		this.setState({
+			searchName: e
+		});
+	}
+
+	// When a user clicks search, handleSearch will activate.
+	handleSearch = () => {
+
+		const playerList: Array<string> = this.props.playerList;
+		const searchName: string = this.state.searchName;
+
+		// Create a filtered list showing user's desired results.
+		let list: Array<string> = [];
+
+		for(const player of playerList) {
+			if(player.includes(searchName)) {
+				list.push(player);
+			}
+		}
+
+		// Set the search result into the state.
+		this.setState({ searchList: list });
+	}
+
+	// Challenge the player chosen.
+	challengePlayer(): void {

```

Comment:
Thanks for the return type :)

You're going to need to pass in which player should be challenged by some sort of unique identifier.